src/game/common/system/radar.h

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief Radar
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+#include "always.h"
+#include "color.h"
+#include "coord.h"
+#include "gametype.h"
+#include "snapshot.h"
+#include "subsysteminterface.h"
+
+class TerrainLogic;
+class RadarObject;
+class GameWindow;
+
+class Radar : public SnapShot, public SubsystemInterface
+{
+public:
+    virtual void CRC_Snapshot(Xfer *xfer) override;
+    virtual void Xfer_Snapshot(Xfer *xfer) override;
+    virtual void Load_Post_Process() override;
+    virtual void Refresh_Terrain(TerrainLogic *logic);
+    virtual void Queue_Terrain_Refresh();
+    virtual void New_Map(TerrainLogic *logic);
+    virtual void Draw(int i1, int i2, int i3, int i4) = 0;
+    virtual void Clear_Shroud() = 0;
+    virtual void Set_Shroud_Level(int i1, int i2, CellShroudStatus status) = 0;
+
+    virtual ~Radar() override;
```

Comment:
Is there a reason destructor is ordered behind? Typically it is put at the top of virtuals list.

Reply:
Destructor is in the second vtable so it makes sense to order it this way.